### DISCUSSION TITLE
Caddy on host - Photoprism in Docker help

### DISCUSSION THREAD

**Discussion by bkev** [photoprism/photoprism#4578] [Q&A]
I'd like a little help if anyone can point me in the right direction.

I have Caddy running on my host and Photoprism running in Docker on that host.

Caddy is running fine and is reverse proxying to various things, including a different Docker container (?) which is working correctly.

But, I cannot get it to work for Photoprism.

My Photoprism is pretty standard and I have the following config in the compose file:

PHOTOPRISM_SITE_URL: "http://localhost:2342/"
    ports:
        - "2342:2342"
    PHOTOPRISM_DISABLE_TLS: "true"   

If I set Caddy to do this

https://hostip:11495 {

reverse_proxy  http://hostip:2342 

}

I get a 502 error.

I've tried changing to this 
PHOTOPRISM_SITE_URL: "http://hostip:2342/"

But I still can't get it to work.

I think the issue is that on the host, if I go to that site it doesn't work, but something else on the network it DOES work for the standard 2342 port. Which is a bit strange maybe?

Is there anyone that could help me figure out what is going wrong?

If I do a docker network ls , I think that the Docker I am able to reverse proxy to is "DRIVER" "host" whereas Photoprism looks like it is "bridge"? Could it be this? 

I'm a little bit lost as I don't know where to look next.

**Comment by bkev** [comment]
I can't seem to see port 2342 if I do a natstat so this is where I think I'm getting stuck. How does the docker network work for me to connect to Photoprism on port 2342 from other devices if it isn't showing here?

**Comment by omerdduran** [comment]
Hello. According to your description, the main problem seems to be related to the Docker network. The PhotoPrism container running on Bridge network cannot be reverse proxied with Caddy.

This should solve your problem.

https://hostip:11495 {
	reverse_proxy host.docker.internal:2342
}


If your problem with Caddy has not been resolved, we recommend that you seek advice from the Caddy community as we cannot provide support for third-party software and services.

**Reply by bkev** [reply]
Thanks for your help.

I can't get that to work and I'm not sure if it's because Caddy isn't running in a Docker of its own.

If I try to go to host.docker.internal:2342 in a browser on the host where Photoprism is running in a Docker, I get

This site can't be reached
Check if there is a typo in host.docker.internal.
DNS_PROBE_FINISHED_NXDOMAIN

If I'm on a physical host (that's Linux) that is running Docker, on that host how can I connect to the Photoprism webpage? I think I'm struggling to just understand how the Docker network connects between the Docker and the host. 

I think if I know or can load up the page on the physical host where Docker is running I will be able to sort Caddy, but I don't understand how to do that or how it works from other devices on the network at the moment.

**Reply by lastzero** [reply]
`host.docker.internal` is a special host for internal Docker networking, AFAIK it only works on Linux. If Caddy is running without Docker on your host and PhotoPrism is listening at port 2342 (with Docker) on all host IP addresses, why don't you tell Caddy to proxy http://localhost:2342/? Why use the external IP address for local port forwarding? 🤔

**Reply by bkev** [reply]
Thanks for the help again.

I think I've figured it out. It's the firewall iptables causing the issue I think. If I turn it off, I can get it to work. When it's on it doesn't. 

Just looking at what I need to add to fix it.

**Reply by bkev** [reply]
Ok so I've need to add this to my firewall rules and now Caddy works and I can get to the site on the host too 

sudo iptables -I INPUT  -s IP of photo prism container -d IP of default gateway on photo prism network -j ACCEPT

**Comment by petergloor** [comment]
I know, it's a little bit late, but maybe this helps someone else. I created a Docker network called "proxy" (it could be named whatever you want) and connected both containers to this network using `--network proxy` in my docker run command.

Then I've set my Caddyfile to something like this:
```
photoprism.example.com {
	# Set this path to your site's directory.
	root * /srv

	# Set up a reverse proxy using photoprism's container name.
	# caddy as well as photoprism must be connected to the same Docker network.
       
	reverse_proxy photoprism:2342
}
```
This worked for me.
